clients/src/main/java/org/apache/kafka/common/network/SaslChannelBuilder.java

```diff
@@ -52,7 +55,15 @@ public SaslChannelBuilder(Mode mode, LoginType loginType, SecurityProtocol secur
     public void configure(Map<String, ?> configs) throws KafkaException {
         try {
             this.configs = configs;
-            this.loginManager = LoginManager.acquireLoginManager(loginType, configs);
+            String mechanism = (String) this.configs.get(SaslConfigs.SASL_MECHANISM);
```

Comment:
This is just a `Map`, so we can't use `getString`. I wanted to change `configure` to take a `Config` type for this reason, but it would break API classes unfortunately.